### PR TITLE
Added debug details

### PR DESCRIPTION
**Pull request by thegecko**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -39,10 +39,10 @@ declare module '@mbed/mbs-plugin' {
 
         /**
          * Registers a handler to return a list of program templates ({@link Template}) for this plugin
-         * Optional {@link targets.Target} ID can be used for restricting templates
+         * Optional {@link targets.Target} can be used for restricting templates
          * @param handler The handler function
          */
-        export function programTemplateHandler(handler: (targetId?: string) => Promise<Template[]>): void;
+        export function programTemplateHandler(handler: (target?: targets.Target) => Promise<Template[]>): void;
```
Comment (arekzaluski) [inline index.d.ts:45]: 👍

File: index.d.ts [hunk 2]
```diff
@@ -77,31 +79,39 @@ declare module '@mbed/mbs-plugin' {
...
+            serialPort?: string;    // An optional serial port the device is connected to
+        }
+
+        /**
+         * Registers a handler to return a {@link ResolvedDevice} for a detected device if known to this plugin
+         * @param criteria The criteria to {@link Match} on
          * @param handler The handler function
          */
-        export function deviceDetectedHandler(criteria: Match[], handler: (device: Device) => Promise<string | undefined>): void;
+        export function deviceDetectedHandler(criteria: Match[], handler: (device: Device) => Promise<ResolvedDevice | undefined>): void;
```
Comment (arekzaluski) [inline index.d.ts:114]: 👍

File: index.d.ts [hunk 3]
```diff
@@ -157,4 +167,41 @@ declare module '@mbed/mbs-plugin' {
...
+            debugTarget?: string;   // The target to use when running/debugging
+            cmsisPack?: string;     // Optional CMSIS pack to use for running/debugging
+        }
+
+        /**
+         * Object describing optional connection details
+         */
+        export interface Connection {
+            connectMode?: 'halt' | 'pre-reset' | 'under-reset' | 'attach';  // The mode to use for connecting to the device
+            eraseMode?: 'sector' | 'chip' | 'mass';                         // The area to erase before running
```
Comment (arekzaluski) [inline index.d.ts:189]: I hope it is generic enough and doesn't apply only to pyOCD.

Comment (thegecko) [reply index.d.ts:189]: I dont have anything to compare it with :(

**Pull request merged**